tests/CMakeLists.txt

```diff
@@ -7,4 +7,5 @@
 add_subdirectory(tpl)
 add_subdirectory(unit)
 add_subdirectory(integration)
-add_subdirectory(tools)
\ No newline at end of file
+add_subdirectory(tools)
+add_subdirectory(applications)
```

Comment:
I think we want this guarded with some CMake option and not included by default?

Reply:
`ENABLE_DEVELOPER_DEFAULTS` perhaps?

Reply:
That's fine with me

Reply:
Done @davidbeckingsale - ready for another review..